pvlib/snow.py

```diff
@@ -185,3 +185,108 @@ def dc_loss_nrel(snow_coverage, num_strings):
        Available at https://www.nrel.gov/docs/fy18osti/67399.pdf
     '''
     return np.ceil(snow_coverage * num_strings) / num_strings
+
+
+def _townsend_Se(S, N):
+    '''
+    Calculates effective snow for a given month based upon the total snowfall
+    received in a month in inches and the number of events where snowfall is
+    greater than 1 inch
+
+    Parameters
+    ----------
+    S : numeric
+        Snowfall in inches received in a month [in]
+
+    N: numeric
+        Number of snowfall events with snowfall > 1" [-]
+
+    Returns
+    -------
+    effective_snowfall : numeric
+        Effective snowfall as defined in the townsend model
+
+    References
+    ----------
+    .. [1] Townsend, Tim & Powers, Loren. (2011). Photovoltaics and snow: An
+       update from two winters of measurements in the SIERRA. Conference
+       Record of the IEEE Photovoltaic Specialists Conference.
+       003231-003236. :doi:`10.1109/PVSC.2011.6186627`
+       Available at https://www.researchgate.net/publication/261042016_Photovoltaics_and_snow_An_update_from_two_winters_of_measurements_in_the_SIERRA
+
+    '''  # noqa: E501
+    return(np.where(N > 0, 0.5 * S * (1 + 1/N), 0))
+
+
+def loss_townsend(snow_total, snow_events, surface_tilt, relative_humidity,
+                  temp_air, poa_global, slant_height, lower_edge_drop_height,
+                  angle_of_repose=40):
+    '''
+    Calculates monthly snow loss based on a generalized monthly snow loss model
+    discussed in [1]_.
+
+    Parameters
+    ----------
+    snow_total : numeric
+        Inches of snow received in the current month. Referred as S in the
+        paper [in]
+
+    snow_events : numeric
+        Number of snowfall events with snowfall > 1". Referred as N in the
+        paper [-]
+
+    surface_tilt : numeric
+        Array surface_tilt [deg]
+
+    relative_humidity : numeric
+        Relative humidity [%]
+
+    temp_air : numeric
+        Ambient temperature [°C]
```

Comment:
I think it's worth explicitly using the word "monthly" for all the monthly inputs, and "average/total/etc" as appropriate.  Temperature is monthly average, and I'm not seeing it in the reference but I guess RH is monthly average as well?